openff/evaluator/layers/equilibration.py

```diff
@@ -82,6 +82,16 @@ class EquilibrationProperty(AttributeClass):
         optional=True,
     )
 
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return False
+
+        cls_attributes = self.get_attributes()
+        for attribute_name in cls_attributes:
+            if getattr(self, attribute_name) != getattr(other, attribute_name):
+                return False
+        return True
```

Comment:
Can this be upstreamed to AttributeClass?

It would be nice to compare objects easily, but I'm not sure how many checks there are in the code for identity vs equality.

Reply:
Probably, but I'm inclined to suggest that we do that in a follow-up